server/routes/api/campaigns.js

```diff
@@ -252,4 +256,160 @@ router.delete(
+      await newStep.save();
+
+      campaign.steps.push(newStep._id);
+      await campaign.save();
+
+      res.json(newStep);
+    } catch (error) {
+      console.log(error);
+      res.status(500).json({ error: "Error adding step" });
+    }
+  }
+);
+
+// @route POST /api/campaigns/:campaignId/steps/:stepId/content
+// @desc Edit the subject and content of a step.
+//       Requires subject and content (stringified editorState) in request body.
+// @access Authenticated Users
+router.post(
+  "/:campaignId/steps/:stepId/content",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    const { campaignId, stepId } = req.params;
+    const { subject, content } = req.body;
+
+    try {
+      const campaign = await Campaign.findById(campaignId);
+
+      if (!campaign) {
+        return res
+          .status(404)
+          .json({ error: `Campaign ${campaignId} not found` });
+      }
+
+      const step = await Step.findById(stepId);
+
+      if (!step) {
+        return res.status(404).json({ error: `Step ${stepId} not found` });
+      }
+
+      const stepInCampaign = campaign.steps.some(curr => curr == stepId);
+
+      if (!stepInCampaign) {
+        return res.status(404).json({
+          error: `Step ${stepId} not found in Campaign ${campaignId}`
+        });
+      }
+
+      step.subject = subject;
+      step.content = content;
+      await step.save();
+
+      res.json(step);
+    } catch (error) {
+      console.log(error);
+      res.status(500).json({ error: "Error updating step content" });
+    }
+  }
+);
+
+// @route POST /api/campaigns/:campaignId/steps/:stepId/prospects
+// @desc Move a selection of prospects in a campaign to one of its steps
+//       Requires prospects (array of prospectIds) in request body.
+// @access Authenticated Users
+router.post(
+  "/:campaignId/steps/:stepId/prospects",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    const { campaignId, stepId } = req.params;
+    const { prospects } = req.body;
+
+    try {
+      const campaign = await Campaign.findById(campaignId);
+
+      if (!campaign) {
+        return res
+          .status(404)
+          .json({ error: `Campaign ${campaignId} not found` });
+      }
+
+      const targetStep = await Step.findById(stepId);
+
+      if (!targetStep) {
+        return res.status(404).json({ error: `Step ${stepId} not found` });
+      }
+
+      const stepInCampaign = campaign.steps.some(curr => curr == stepId);
+
+      if (!stepInCampaign) {
+        return res.status(404).json({
+          error: `Step ${stepId} not found in Campaign ${campaignId}`
+        });
+      }
+
+      for (let i = 0; i < prospects.length; i++) {
+        const prospectObject = campaign.prospects.find(
+          curr => curr.prospectId == prospects[i]
+        );
+        if (!prospectObject) {
+          continue;
+        }
+
+        //If the prospect was in another step, remove it from that step
```

Comment:
I don't think we need to do this. My understanding is prospects can be in many steps at once, so we can check their status at each step, for example. @skassam21 correct me if I'm wrong

Reply:
This has been removed as per our discussion.